packages/commerce-sdk-react/src/hooks/ShopperBaskets/mutation.ts

```diff
@@ -169,46 +171,236 @@ export const ShopperBasketsMutations = {
     AddTaxesForBasket: 'addTaxesForBasket'
 } as const
 
-type ShopperBasketMutationType = typeof ShopperBasketsMutations[keyof typeof ShopperBasketsMutations]
+export type ShopperBasketMutationType = typeof ShopperBasketsMutations[keyof typeof ShopperBasketsMutations]
+
+export const getQueryKeysMatrix = (customerId: string | null) => {
+    const updateBasketQuery = (basketId?: string) => {
+        // TODO: we're missing headers, rawResponse -> not only {basketId}
+        const arg = {basketId}
+        return basketId
+            ? {
+                  update: [['/baskets', basketId, arg]]
+              }
+            : {}
+    }
+
+    const removeBasketQuery = (basketId?: string) => {
+        const arg = {basketId}
+        return basketId
+            ? {
+                  remove: [['/baskets', basketId, arg]]
+              }
+            : {}
+    }
+
+    const invalidateCustomerBasketsQuery = (customerId: string | null) => {
+        // NOTE: need arg here too actually because the query key will be used later in the test
+        // TODO: actually... it depends on how the test is done
+        // TODO: maybe close the ticket https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001EWO8MYAX/view
+        const arg = {customerId}
+        return customerId
+            ? {
+                  invalidate: [['/customers', customerId, '/baskets', arg]]
+              }
+            : {}
+    }
+
+    return {
+        addCouponToBasket: (
+            params: Argument<Client['addCouponToBasket']>,
+            response: DataType<Client['addCouponToBasket']>
+        ): QueryKeysMatrixElement => {
+            const basketId = params.parameters.basketId
+
+            return {
+                ...updateBasketQuery(basketId),
+                ...invalidateCustomerBasketsQuery(customerId)
+            }
```

Comment:
Filling out the query keys matrix can be time-consuming and error prone. So whenever possible, I'd like to extract the repeating code into their own functions like these.